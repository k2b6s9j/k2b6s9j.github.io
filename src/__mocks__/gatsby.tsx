import * as React from "react";
const gatsby = jest.requireActual("gatsby");
import { graphql } from "graphql";

type MockedGraphQLReturn = Promise<any>;

const mockedGraphQLQuery = (query: string): MockedGraphQLReturn =>
  graphql(undefined, query);

interface StaticQueryHOCProps {
  query: MockedGraphQLReturn;
  render?: (data: object) => React.ReactElement;
  children?: (data: object) => React.ReactElement;
}

const mockedStaticQueryHOC = (
  props: StaticQueryHOCProps
): React.ReactElement => {
  if (!props.children || !props.render) {
    throw new Error("Render prop must be provided!");
  }

  return (
    (props.children && props.children({})) ||
    (props.render && props.render({})) || <div>Error!</div>
  );
};

const mockedStaticQueryHook = (data: MockedGraphQLReturn): object => ({});

module.exports = {
  ...gatsby,
  graphql: jest.fn(mockedGraphQLQuery),
  Link: jest.fn(
    ({
      // these props are invalid for an `a` tag
      /* eslint-disable @typescript-eslint/no-unused-vars */
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      /* eslint-enable @typescript-eslint/no-unused-vars */
      to,
      ...rest
    }): React.ReactElement => <a {...rest} href={to} />
  ),
  StaticQuery: jest.fn(mockedStaticQueryHOC),
  useStaticQuery: jest.fn(mockedStaticQueryHook)
};
