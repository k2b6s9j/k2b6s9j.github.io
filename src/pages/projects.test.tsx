import * as React from "react";
import renderer from "react-test-renderer";
import { useStaticQuery } from "gatsby";
import Page from "./projects";

beforeEach((): void => {
  (useStaticQuery as jest.Mock).mockImplementation((): object => ({
    file: {
      childImageSharp: {
        fixed: {
          base64: "",
          width: 100,
          height: 100,
          src: "",
          srcSet: ""
        }
      }
    },
    backdrop: {
      childImageSharp: {
        fluid: {
          base64: "",
          aspectRatio: 1.333,
          src: "",
          srcSet: "",
          srcWebp: "",
          srcSetWebp: "",
          sizes: ""
        }
      }
    },
    site: {
      siteMetadata: {
        title: "",
        description: "",
        siteUrl: "",
        twitterUsername: "",
        instagramUsername: "",
        linkedinUsername: "",
        githubUsername: ""
      }
    }
  }));
});

describe("Projects Page", (): void => {
  it("renders correctly", (): void => {
    const tree = renderer
      .create(
        <Page
          data={{
            allFile: {
              edges: [
                {
                  node: {
                    childMarkdownRemark: {
                      id: "",
                      rawMarkdownBody: "",
                      excerpt: "",
                      fields: {
                        slug: ""
                      },
                      frontmatter: {
                        title: "",
                        description: ""
                      }
                    }
                  }
                }
              ]
            }
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
