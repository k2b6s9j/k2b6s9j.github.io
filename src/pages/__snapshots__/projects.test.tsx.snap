// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects Page renders correctly 1`] = `
Array [
  .emotion-7 {
  height: 3rem;
  width: 100vw;
  margin-bottom: 1em;
}

.emotion-6 {
  height: 3rem;
  max-width: 100vw;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-backdrop-filter: blur(5px);
  backdrop-filter: blur(5px);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 1rem;
  padding-right: 1rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-1 {
  text-align: start;
  max-width: 50%;
}

.emotion-0 {
  font-weight: 600;
  color: #141414;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-5 {
  text-align: end;
  max-width: 50%;
}

.emotion-5 > :not(:last-child) {
  margin-right: 1em;
}

<head>
    Object {}
    Object {
      "lang": "en",
    }
    <meta
      content="Kepler Sticka-Jones"
      data-react-helmet={true}
      name="author"
    />
    <meta
      content=""
      data-react-helmet={true}
      name="description"
    />
    <meta
      content="#409040"
      data-react-helmet={true}
      name="theme-color"
    />
    <meta
      content="Projects | "
      data-react-helmet={true}
      property="og:title"
    />
    <meta
      content="website"
      data-react-helmet={true}
      property="og:type"
    />
    <meta
      content=""
      data-react-helmet={true}
      property="og:url"
    />
    <meta
      content=""
      data-react-helmet={true}
      property="og:image"
    />
    <meta
      content="image/jpeg"
      data-react-helmet={true}
      property="og:image:type"
    />
    <meta
      content="480"
      data-react-helmet={true}
      property="og:image:width"
    />
    <meta
      content="480"
      data-react-helmet={true}
      property="og:image:height"
    />
    <meta
      content=""
      data-react-helmet={true}
      property="og:description"
    />
    <meta
      content="summary"
      data-react-helmet={true}
      name="twitter:card"
    />
    <meta
      content="@"
      data-react-helmet={true}
      name="twitter:site"
    />
    <meta
      content="@"
      data-react-helmet={true}
      name="twitter:creator"
    />
    <meta
      content="E5D802A305BE2DD99EB4484A4B1DD90E"
      data-react-helmet={true}
      name="msvalidate.01"
    />
    <script
      data-react-helmet={true}
      type="application/ld+json"
    >
      Object {
      "@context": "http://www.schema.org",
      "@type": "CollectionPage",
      "about": Object {
        "@type": "ItemList",
        "itemListElement": Array [
          Object {
            "@type": "Thing",
            "description": "",
            "name": "",
            "url": "",
          },
        ],
        "name": "Projects | Kepler Sticka-Jones",
        "numberOfItems": 1,
        "url": "/projects",
      },
      "breadcrumb": Object {
        "@type": "BreadcrumbList",
        "itemListElement": Array [
          Object {
            "@type": "ListItem",
            "item": Object {
              "@id": "https://keplersj.com/",
              "name": "Kepler Sticka-Jones",
            },
            "position": 1,
          },
          Object {
            "@type": "ListItem",
            "item": Object {
              "@id": "https://keplersj.com/projects/",
              "name": "Projects",
            },
            "position": 2,
          },
        ],
      },
      "name": "Projects | Kepler Sticka-Jones",
      "url": "/projects",
    }
    </script>
    <title
      data-react-helmet={true}
    >
      Projects | 
    </title>
  </head>

<div
    className="emotion-7 gatsby-image-wrapper"
    style={
      Object {
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "opacity": 0.99,
        "position": "relative",
      }
    }
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
          .emotion-7:before,
          .emotion-7:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            background-color: #040e18;
            transition: opacity 0.5s ease 0.25s;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .emotion-7:before {
            z-index: -100;
            
            
            opacity: 0; 
          }
          .emotion-7:after {
            z-index: -101;
            
            
          }
        ",
        }
      }
    />
    <noscript>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": "
          .emotion-7:before {
            opacity: 1;
            background-image: ;
          }",
          }
        }
      />
    </noscript>
    <nav
      className="emotion-6"
    >
      <div
        className="emotion-1"
      >
        <a
          className="emotion-0"
          href="/"
        >
          Kepler Sticka-Jones
        </a>
      </div>
      <div
        className="emotion-5"
      >
        <a
          className="emotion-0"
          href="/blog"
        >
          Blog
        </a>
        <a
          className="emotion-0"
          href="/projects"
        >
          Projects
        </a>
        <a
          className="emotion-0"
          href="/contact"
        >
          Contact
        </a>
      </div>
    </nav>
  </div>,
  .emotion-1 {
  max-width: 55em;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:55em) {
  .emotion-1 {
    margin-left: 2em;
    margin-right: 2em;
  }
}

.emotion-0:not(:last-child) {
  margin-bottom: 1.5rem;
}

<main>
    <div
      className="emotion-1"
    >
      <h1>
        Projects
      </h1>
      <div>
        <div
          className="emotion-0"
        >
          <h2>
            
          </h2>
          <div>
            
          </div>
        </div>
      </div>
    </div>
  </main>,
]
`;
